### PR TITLE
Track audiobooks kept in a Plex Music library as books

### LINKED ISSUE #1050: [Feature Request] Custom import for custom library (aka: audiobooks in Plex Music library)
**Describe the solution you'd like**  
Could you add possibility to create "custom" tracking library, which does not try to match to some online system. Just imports and track data from Plex library with information there is (or any other sources supproted).

Like: "Create custom Mediatype" in this case "Audiobooks", and then import that from selected library, like in my case Plex Audiobook library, which is set as Music library in Plex.. If I try to import it now to "music" it does not import anything, as it does not find them in that stupid MusciBrainz... It is not every time for everything needed it to be matched to some online system... whole library is as "music" library, as books in there are NOT English amazon books.... and there is no online match service which could be used to match them.. all are manually generated info with MP3 tags, so library works. It would be really nice to add it to trackers here... (and in case I switch to some _other_ audiobook service.. well match that then to there, and keep track where Plex was left...)

**Describe alternatives you've considered**  
Not sure you can do any alternative for these kind of cases.. as current approach "only track libraries for something that can be matched to some 3rd party system" does _not_ always work.. as media does not exist in those..

**Additional context**  
Hope this could be implemented easily and make this more versatile :)

### PR DESCRIPTION
**Pull request by dannyvfilms**
## Summary

Audiobooks shelved in a Plex **Music** library now import as books instead of falling through the music path.

Plex has no audiobook media type for music sections. A Plex Music library is `type="artist"`, and its rows arrive as `type="track"` with the author in `grandparentTitle`, the book in `parentTitle`, and the chapter in `title`. Floppy has to decide for itself that an album is a book — Plex will never say so.

Today those rows go to `_process_music_entry`, which returns early when music tracking is disabled (nothing imports at all), and otherwise fuzzy-searches MusicBrainz for "Chapter 7". So a hand-tagged, non-English audiobook library imports as nothing, or as thousands of junk track rows.

What changed:

- **New `Sources.PLEX`** with a local-only provider shim, mirroring the existing Audiobookshelf and Storyteller book sources: synthetic `media_id`, no external match, `format="audiobook"`, progress in minutes. `plex` was already in `SOURCES_CONFIG` with a logo, so the source chip needed nothing new.
- **Detection** (`integrations/imports/plex_audiobooks.py`) scores each album on signals Plex actually provides: genre tags, the library's metadata agent (Audnexus/Booksonic are conclusive), chapter-style track titles, chapter-length durations, and the absence of a MusicBrainz GUID — which real music matched by Plex's own music agent always carries. Pure functions, no I/O, so the thresholds are tunable against fixtures. The section fetch now also keeps `agent`/`scanner`, which it was discarding.
- **A per-library Auto / Music / Audiobooks setting**, shown in the import form only for music libraries and pre-selected to Audiobooks when the library already looks like one. Stored on `PlexAccount.section_settings` so scheduled runs and the live webhook read the same choice.
- **The importer groups chapters by album**, so one album upserts one book with runtime and progress from the album's tracks, rather than one row per chapter. Auto-detected albums are named in the import summary so an over-eager heuristic is visible. Audiobooks found while book tracking is off now warn instead of vanishing silently — that silence is the reported symptom.
- **The webhook** routes flagged libraries to the same shared upsert (extracted to `integrations/plex_audiobook_sync.py` so importer and webhook can't drift), gated behind a cheap per-track pre-filter so an ordinary song never costs an album lookup. When a library is deliberately kept as Music, audiobook-shaped tracks now skip the MusicBrainz search that produced the junk matches.

Plex art requires an `X-Plex-Token`, so covers go through a signed proxy rather than storing the token in `Item.image` or emitting it into an `` — the same pattern as the existing Audiobookshelf cover proxy.

Deliberately **not** built: arbitrary user-defined media types. Audiobooks already have first-class support in Floppy (`format="audiobook"`, minute-based progress, smart-rule labels), so this connects Plex to that rather than adding a parallel type system.

### Visual changes

Verified in a local browser run; screenshots can be attached on request.

**Book detail page** for an imported Plex audiobook — resolves entirely from local Plex metadata, no book provider involved:

| Field | Value |
| --- | --- |
| Progress | `1h 00min` (time, not a page count) |
| Author | Franz Kafka (from the Plex artist) |
| Format | audiobook |
| Runtime minutes | 120 (summed from the album's chapters) |
| Genres | the album's Plex genre tag |
| Synopsis | the album summary |

**Import modal** gains a *Content type* selector directly under Library, with Auto-detect / Music / Audiobooks and a line of helper text. Confirmed behavior:

- music library, no audiobook signals → visible, defaults to `auto`
- library on the Audnexus agent → visible, pre-selected to `audiobook`
- movie library → hidden entirely

## Unrelated CI repairs also in this branch

`test (3.12)` had been failing on `latest` since run #1201 — every run from #1202 onward, including this PR's base commit. Four of the seven commits here fix that; none of it is caused by the feature, and reviewers may prefer it split out.

1. **`87e2c38` — a real product bug.** `metadata_default_source()` falls back to the first *available* source when the chosen one is disabled. Anime's source order is `[MAL, TMDB, TVDB]`, so a user who picked **TVDB without an API key** fell all the way back to **MAL**. For anime the provider decides the library's *storage shape* (MAL = flat per-cour `Anime` rows, TMDB/TVDB = grouped TV-shaped rows), so that silently converted a grouped anime library to flat. `prefers_grouped_anime()`'s docstring already assumed the fallback stayed grouped; now it does.
2. **`0cc0d73`** — two `lists` Playwright tests mock the MAL provider and assert on a MAL-sourced item, but `users/0129` changed the Anime default to TMDB, so a fresh test user searched anime through TMDB (unreachable in tests) and rendered "No results found". Pinned those users to MAL. Also made the session-history calendar test wait for the deferred `fragment=secondary` htmx load before opening the modal, reusing the `expect_response` idiom already in that file.
3. **`75ce730` → `2ea3411`** — the card's Lists action click was intercepted by the poster image once it loaded. A hover-first attempt failed too (CI rejected the hover with the same "subtree intercepts pointer events"), so the helper now dispatches the click on the button directly. **This trades pointer fidelity for determinism in that one helper**; every assertion in both tests is unchanged.

Worth a separate look, unrelated to this PR: if a real pointer cannot reach `.media-card-overlay` on the Chromium CI runs, the media card's hover actions may be broken for users on that browser. `.media-card-overlay` carries `z-index: 5` and sits after the poster in the DOM, so it should paint above it. I could not investigate properly — the only Chromium available in my sandbox is v141 against CI's v151, and it does not reproduce.

## AI Assistance

`claude-opus-5`

## How It Was Tested

- `pytest src/integrations/tests/imports/test_plex_audiobooks.py` -> 13 passed (scoring fixtures: genre-tagged audiobook, untagged audiobook detected by shape, real music album with MBIDs scoring low, spoken-word music edge case, section-hint behavior)
- `pytest src/integrations/tests/test_plex_audiobook_import.py` -> 11 passed (forced routing, auto-detection, one book per album, idempotent re-import advancing progress, disabled-book-tracking warning, music album still importing as music, cross-server cache scoping)
- `pytest src/integrations/tests/test_webhooks_plex_audiobooks.py` -> 9 passed (flagged library updates the book, later chapters advance it, normal songs never reach the album lookup, Music choice skips the MusicBrainz search, wrong-server album fetch, shared-server token, section hint)
- `pytest src/integrations/tests/test_webhooks_plex.py` -> 62 passed
- `pytest src/integrations/tests/test_plex_import_logic.py src/integrations/tests/imports/test_audiobookshelf.py src/app/tests/providers/test_services.py src/integrations/tests/test_plex_list_sections.py` -> 139 passed
- **Full suite, exactly as CI runs it** (`manage.py test app users integrations lists events api config --exclude-tag network`) -> `Ran 4994 tests ... OK (skipped=6)`
- `ruff check src/app src/integrations src/users` -> clean; `manage.py check` -> no issues
- `manage.py migrate` from an empty database -> both migrations apply; `makemigrations --check` -> no drift
- Manual QA in a local run: created a Plex audiobook through the real upsert path and confirmed the book detail page and the import modal's Content type selector render correctly across music / audiobook-hinted / movie libraries.

A test caught a real bug during development: the German plural "Hörbücher" folds to `hörbüch`, which the hint `hörbuch` never matched. Both sides are now diacritic-folded, which handles the whole class rather than that one word.

## Public API & Documentation Handoff

- [x] Domain terminology is up to date (`python -m app.domain_vocabulary --check` -> clean)
- [x] Not applicable (no API endpoints changed; the one new route is the authenticated cover proxy)

## Human Review & Quality Assurance

- [ ] Code review completed
- [x] Visual or manual QA verified (browser testing via the `run-floppy` skill)

## Database & Migration Safety

- [x] No existing or shared migrations were altered or renumbered
- [x] Migration hygiene passed (`uv run --no-sync python src/manage.py check_migration_hygiene --strict` -> passed; both new migrations use the guarded `AddFieldIfNotExists` / `AddConstraintIfNotExists` / `RemoveConstraintIfExists` operations)
- [ ] Not applicable

Two new migrations: `app/0173_alter_item_source_add_plex` (adds `plex` to the source choices and the `app_item_source_valid` constraint, in the established shape of `0127`/`0170`) and `integrations/0032_plexaccount_section_settings`.

## Related Issues

Fixes #1050

## Note for reviewers

Detection quality depends on how the user tagged their files. The thresholds are calibrated against fixtures, not real-world libraries, so they will likely need tuning once the issue reporter tries it — which is exactly why the explicit per-library override exists and why the scoring lives in an isolated, fixture-driven module.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01YbLF56YNtDze9kxVhb6Wsf

### CODE REVIEW

File: src/integrations/webhooks/plex.py [hunk 1]
```diff
@@ -1302,6 +1313,115 @@ def _extract_uuid(self, value):
...
+        """Return (album metadata, tracks) for the played track's album."""
+        metadata = payload.get("Metadata", {}) or {}
+        album_key = metadata.get("parentRatingKey") or metadata.get("parentKey")
+        if not album_key:
+            return None, []
+        album_key = str(album_key).rsplit("/", 1)[-1]
+
+        plex_account, plex_uri = self._resolve_plex_server(user, payload)
+        if not plex_account or not plex_uri:
+            return None, []
```
[the remark this answers is not in the document]
Comment (dannyvfilms) [reply src/integrations/webhooks/plex.py]: Confirmed and fixed in f5ac76b.

A real Plex webhook's `Server` block is `{title, uuid}` with no `uri`, so `_resolve_plex_server` never got a URI from the payload and always fell through to the "first cached section" branch — not just occasionally, but on every audiobook event. You're also right about the token: it returned `plex_account.plex_token` rather than the section's `access_token`, so a server shared by another user would have failed auth.

`_fetch_audiobook_album` now resolves the section by `Server.uuid` (plus `librarySectionID`) through a new shared `plex.connection_for_machine()`, preferring the section's own access token and falling back to the old path only when no cached section matches. The cover proxy in `integrations/views.py` had an inlined copy of this lookup, so it now uses the shared helper too.

Two regression tests cover it: `test_album_is_fetched_from_the_payloads_own_server` puts a decoy section from another server first in the cache and asserts the fetch still goes to the payload's server, and `test_shared_server_uses_the_sections_access_token` asserts the section token is used.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/integrations/imports/plex.py [hunk 2]
```diff
@@ -645,6 +691,111 @@ def _process_entry(self, entry: dict, uri: str, section_type: str | None = None)
...
+
+        album_key = self._album_key(metadata)
+        if not album_key:
+            return self._current_section_content_kind == (
+                plex_audiobooks.CONTENT_KIND_AUDIOBOOK
+            )
+
+        album_key = str(album_key).rsplit("/", 1)[-1]
+        if album_key in self._audiobook_album_verdicts:
+            return self._audiobook_album_verdicts[album_key]
```
[the remark this answers is not in the document]
Comment (dannyvfilms) [reply src/integrations/imports/plex.py]: Confirmed and fixed in f5ac76b.

Plex rating keys are small integers scoped to a server, so a collision across two servers in an `all` import isn't a corner case — it's likely. Both caches are now keyed by `(machine identifier, rating key)` via a single `_album_cache_key()` helper, which also normalizes the `/library/metadata/100` path form of `parentKey` in one place rather than at each call site.

`test_same_rating_key_on_two_servers_is_classified_separately` covers it: the same album key is classified under one server as an audiobook and under another as music, and the verdict cache is asserted to hold only the first server's entry.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/integrations/webhooks/plex.py [hunk 3]
```diff
@@ -1302,6 +1313,115 @@ def _extract_uuid(self, value):
...
+        if kind == plex_audiobooks.CONTENT_KIND_MUSIC:
+            return "music_no_lookup"
+        return "book" if self._confirm_audiobook_album(payload, user) else None
+
+    def _confirm_audiobook_album(self, payload, user):
+        """Confirm an auto-detected audiobook by scoring its whole album."""
+        album, tracks = self._fetch_audiobook_album(payload, user)
+        if not album:
+            return False
+        return plex_audiobooks.is_audiobook_album(album, tracks)
```
[the remark this answers is not in the document]
Comment (dannyvfilms) [reply src/integrations/webhooks/plex.py]: Confirmed and fixed in f5ac76b.

The split you describe is exactly what would have happened: the two paths would have disagreed on the same album, so a book imported from history would then collect its live scrobbles as music.

`_confirm_audiobook_album` now resolves the cached section for the payload's server and library and passes the same `section_hint` the importer uses, via `_section_audiobook_hint()`. `test_section_hint_is_applied_to_live_detection` uses a borderline album (three ~25-minute "Part N" tracks) that scores below threshold on its own — the existing unit test `test_section_hint_carries_a_borderline_album` pins that — and asserts it is now detected in a library named "Audiobooks".

---
_Generated by [Claude Code](https://claude.ai/code)_
